math/n_bonacci.cpp

```diff
@@ -0,0 +1,58 @@
+/**
+ * @file N-bonacci.cpp
+ * @brief Implementation of N-bonacci Series
```

Comment:
Provide a Wikipedia link in Markdown format (if possible).
If there's no Wikipedia link, use another web source for algorithm explanation. 🙂

Reply:
http://oeis.org/wiki/N-bonacci_numbers
please refer to above link :)